packages/authentication/src/types.ts

```diff
@@ -110,3 +139,28 @@ export function registerAuthenticationStrategy(
     },
   );
 }
+
+/**
+ * Registers a security scheme spec as an extension of the @loopback/openapi-v3
+ * OAS_ENHANCER_EXTENSION_POINT_NAME extension point.
+ *
+ * @param context - Context object
+ * @param name - Prefix for security scheme spec enhancer name
+ * @param securitySchemes - OAS security scheme objects
+ */
+export function registerAuthSecuritySpec(
```

Comment:
This register function looks reasonable to me 👍 
Next we could add a few acceptance test to verify if a security spec get merged when call `registerAuthenticationStrategy(this, strategyName);`

Reply:
And I need to double check if we register strategy in the app constructor, does it happen before or after the rest server gets instantiated.